src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(Storage storage, FruitTransaction transaction) throws RuntimeException {
+            String fruitName = transaction.getFruit();
+            int oldQuantity = storage.get(fruitName);
```

Comment:
Ensure that the 'get' method on the storage doesn't return a null value which could cause a NullPointerException when unboxing the Integer to an int. It would be safer to handle a potential null value here.